src/nbl/ui/CWindowX11.cpp

```diff
@@ -227,6 +230,84 @@ CWindowX11::CWindowX11(core::smart_refctd_ptr<system::ISystem>&& sys, uint32_t _
 
     m_dpy = dpy;
     m_native = win;
+
+    XContext classId = XUniqueContext();
+    XSaveContext(m_dpy, m_native, classId, (XPointer)this);
+}
+
+void CWindowX11::processEvent(XEvent event)
+{
+    IEventCallback eventCallback = getEventCallback();
+
+    switch(event.type)
+    {
+        case ConfigureNotify:
+        {
+            XConfigureEvent e = event.xconfigure;
+            // Resized
+            if(e.width != m_width || e.height != m_height)
+            {
+                eventCallback->onWindowResized(this, e.width, e.height);
+                x11.pXResizeWindow(m_dpy, m_native, e.width, e.height);
+
+            }
+            // Moved
+            if(e.x != m_x || e.y != m_y)
+            {
+                eventCallback->onWindowMoved(this, e.x, e.y);
+                x11.pXMoveWindow(m_dpy, m_native, e.x, e.y);
```

Comment:
I think that `onWindowXYZ` should return a bool to let us know if we're allowed to proceed with the request

Reply:
I what cases will it return false?

Reply:
when you're not allowing the backend to move the window (no call to `x11.pXMoveWindow`)

Reply:
TODO @sadiuk (might want to coordinate with @achalpandeyy to make an example that allows swapchain resizing)